core/controlplane/config/config.go

```diff
@@ -817,6 +820,7 @@ type KubeDnsAutoscaler struct {
 }
 
 type KubeDns struct {
```

Comment:
Yeah, this should retain the name for backward compatibility for now.
But how about renaming this to `ClusterLocalDns` in the **future**?
`KubeDns` sounds like literally specific to `kube-dns`, that is just one of cluster local DNS server implementations available to Kubernetes.